src/main/java/engine/move/PromoteMove.java

```diff
@@ -0,0 +1,60 @@
+package engine.move;
+
+import engine.entities.Piece;
+import engine.entities.Board;
+
+/**
+ * A representation of a (pawn) promotion move.
+ */
+public class PromoteMove extends Move {
+    private String promotedType;
+
+    /**
+     * Creates a PromoteMove object with the given coordinate pairs.
+     * The provided string is the type the moved piece should have.
+     *
+     * @param origin      the origin of the piece to be moved
+     * @param destination the destination of the piece to be moved
+     */
+    public PromoteMove(int[] origin, int[] destination, String promotedType) {
+        super(origin, destination);
+        this.promotedType = promotedType;
+    }
+
+    /**
+     * Mutates the given board (as well as players and pieces within the board)
+     * to reflect the state of the board after this move is made.
+     * The moved piece should be replaced with a piece of the new type.
+     *
+     * @param board the board on which this move should be executed
+     */
+    @Override
+    public void execute(Board board) {
+        Piece pawn = board.getPieceAtAbsCoords(origin[0], origin[1]);
+        // (No captures)
+        // Move pawn to row
+        movePiece(board, origin, destination);
+        // Promote to new type
+        pawn.setType(promotedType);
```

Comment:
This will not properly work because the point value is not changed. Should write a test to cover this case. An appropriate fix is to create a brand new Piece object instead of trying to mutate the pawn in place.